src/ansi-c/ansi_c_entry_point.cpp

```diff
@@ -268,12 +268,7 @@ bool generate_ansi_c_start_function(
         argc_symbol.is_lvalue = true;
         argc_symbol.mode = ID_C;
 
-        if(!symbol_table.insert(std::move(argc_symbol)).second)
-        {
-          messaget message(message_handler);
-          message.error() << "failed to insert argc symbol" << messaget::eom;
-          return true;
-        }
+        symbol_table.insert(std::move(argc_symbol));
```

Comment:
Shouldn't there at least be some type compatibility check?

Reply:
Where should it be done?

Reply:
I'd say here is a good place? It will require some more code, but it would be nice to check whether argc/argv are present already and, if so, their types match expectations. Else we should continue to fail.

Reply:
I have added checks that `argc` and `argv` are identical to the required symbols.